sentry/src/test/java/io/sentry/protocol/SerializationUtils.kt

```diff
@@ -0,0 +1,38 @@
+package io.sentry.protocol
+
+import io.sentry.FileFromResources
+import io.sentry.ILogger
+import io.sentry.JsonDeserializer
+import io.sentry.JsonObjectReader
+import io.sentry.JsonObjectWriter
+import io.sentry.JsonSerializable
+import java.io.StringReader
+import java.io.StringWriter
+
+class SerializationUtils {
+    // TODO: refactor every ser and deser tests to reuse this utils, a lot of boilerplate
```

Comment:
There's a lot of duplication in the tests that can be refactored, not doing in this PR, but I've refactored the classes I've touched at least.